store/clientSettings.ts

```diff
@@ -2,48 +2,79 @@ import { MutationTree, ActionTree } from 'vuex';
 import nuxtConfig from '~/nuxt.config';
 
 /**
- * Casted typings for the CustomPrefs property of DisplayPreferencesDto
+ * Mutations that should be synced to the server
  */
-export interface CustomPreferences {
+export const syncedMutations = ['SET_DARK_MODE', 'SET_LOCALE'];
+
+/**
+ * Cast typings for the CustomPrefs property of DisplayPreferencesDto
+ */
+export interface ClientSettingsState {
   darkMode: boolean;
   locale: string;
+  lastSync: number | null;
 }
 
-export const defaultState = (): CustomPreferences => ({
+export const getDefaultState = (): ClientSettingsState => ({
   darkMode:
     nuxtConfig.vuetify?.theme?.dark !== undefined
       ? nuxtConfig.vuetify?.theme?.dark
       : true,
-  locale: 'auto'
+  locale: 'auto',
+  lastSync: null
```

Comment:
This should live in the root store so we can directly save all the state without further processing in the plugin.

Reply:
That wouldn't work with multiple preference stores, though. We're not going to save every one of them each time one changes.

We'll also likely have an indicator for file syncing, which would also fall under this name. It makes more sense to keep these timestamps in the store they're related to.